src/compiler/types.ts

```diff
@@ -4329,6 +4330,7 @@ namespace ts {
         Iterator = 1 << 23,                     // The expression to a `yield*` should be treated as an Iterator when down-leveling, not an Iterable.
         NoAsciiEscaping = 1 << 24,              // When synthesizing nodes that lack an original node or textSourceNode, we want to write the text on the node with ASCII escaping substitutions.
         /*@internal*/ TypeScriptClassWrapper = 1 << 25, // The node is an IIFE class wrapper created by the ts transform.
+        /*@internal*/ NoNameGenerationScope = 1 << 26, // Used to indicate when a function or other wrapper shouldn't be considered a name generation scope because its generated locals are merged with a higher scope

```

Comment:
We already have `ReuseTempVariableScope` which is supposed to mean the same thing. I like your approach where you added `node` to `pushNameGenerationScope` and `popNameGenerationScope`. Perhaps we can combine the two approaches and deduplicate the behavior in the emitter?